contracts/utils/structs/EnumerableSet.sol

```diff
@@ -238,6 +245,20 @@ library EnumerableSet {
         return address(uint160(uint256(_at(set._inner, index))));
     }
 
+    /**
+     * @dev Return the entier set in an array
+     */
+    function values(AddressSet storage set) internal view returns (address[] memory) {
+        bytes32[] storage store = set._inner._values;
+        address[] storage result;
+
+        assembly {
+            result.slot := store.slot
+        }
+
+        return result;
```

Comment:
If you're returning `memory` arrays and not storage ones, this is how you'd cast a memory array:

```suggestion
        bytes32[] memory store = _values();
        address[] memory result;

        assembly {
            result := store
        }

        return result;
```

This is essentially a pointer copy.

Reply:
I didn't know you could do a memory array cast like that, definitelly sounds like the right solution !